test/integration/epp/hermetic_test.go

```diff
@@ -1690,18 +1690,6 @@ func BeforeSuite() func() {
 		logutil.Fatal(logger, err, "Can't read object manifests", "path", manifestsPath)
 	}
 
-	for _, doc := range docs {
```

Comment:
looks like the first loop the is unmarshalling `InferenceModel` objects while the second loop the handles `InferencePool` objects.

I agree there is no need for running the loop twice.

you can simplify by using Unstructured object.